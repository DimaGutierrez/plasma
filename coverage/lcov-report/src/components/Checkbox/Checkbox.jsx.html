<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Checkbox/Checkbox.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Checkbox</a> Checkbox.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { uniqueId } from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import style from './style.scss';
import { getDataAttrs, getDataProps } from '../../dataUtils';
&nbsp;
class Checkbox extends React.Component {
  render() {
    const {
      isBold,
      checked,
      indeterminate,
      name,
      text,
      onChange,
      disabled,
      value,
      data,
      onBlur,
    } = this.props;
    const id = uniqueId('id');
    const inputClassName = indeterminate ? <span class="branch-0 cbranch-no" title="branch not covered" >style.indeterminate </span>: style.original;
    const wrapperClassName = cx(style.wrapper, {
      [style.wrapperDisabled]: disabled,
    });
    const textClassName = cx(style.text, {
      [style.textBold]: isBold,
      [style.textDisabled]: disabled,
    });
&nbsp;
    return (
      &lt;div {...getDataAttrs(data)}&gt;
        &lt;div&gt;
          &lt;label htmlFor={id} className={wrapperClassName}&gt;
            &lt;input
              disabled={disabled}
              checked={checked || value}
              className={inputClassName}
              type="checkbox"
              id={id}
              name={name}
              onChange={onChange}
              onBlur={onBlur}
            /&gt;
            &lt;div className={style.checkbox} /&gt;
            &lt;div className={textClassName}&gt;{text}&lt;/div&gt;
          &lt;/label&gt;
        &lt;/div&gt;
        {this.props.description &amp;&amp; (
<span class="branch-1 cbranch-no" title="branch not covered" >          &lt;span className={style.description}&gt;{this.props.description}&lt;/span&gt;</span>
        )}
      &lt;/div&gt;
    );
  }
}
&nbsp;
Checkbox.defaultProps = {
  checked: false,
  value: false,
  indeterminate: false,
  text: 'Option',
  name: 'checkbox',
};
&nbsp;
Checkbox.propTypes = {
  checked: PropTypes.bool,
  value: PropTypes.bool,
  indeterminate: PropTypes.bool,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.element]).isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  description: PropTypes.string,
  onBlur: PropTypes.func,
  ...getDataProps(),
};
&nbsp;
Checkbox.displayName = 'Plasma@Checkbox';
&nbsp;
export default Checkbox;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 29 2018 15:55:13 GMT-0400 (EDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
