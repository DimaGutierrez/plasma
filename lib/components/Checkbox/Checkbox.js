!(function(e, t) {
  for (var n in t) e[n] = t[n];
})(
  exports,
  (function(e) {
    function t(r) {
      if (n[r]) return n[r].exports;
      var o = (n[r] = { i: r, l: !1, exports: {} });
      return e[r].call(o.exports, o, o.exports, t), (o.l = !0), o.exports;
    }
    var n = {};
    return (
      (t.m = e),
      (t.c = n),
      (t.d = function(e, n, r) {
        t.o(e, n) || Object.defineProperty(e, n, { configurable: !1, enumerable: !0, get: r });
      }),
      (t.n = function(e) {
        var n =
          e && e.__esModule
            ? function() {
                return e.default;
              }
            : function() {
                return e;
              };
        return t.d(n, 'a', n), n;
      }),
      (t.o = function(e, t) {
        return Object.prototype.hasOwnProperty.call(e, t);
      }),
      (t.p = ''),
      t((t.s = 60))
    );
  })({
    0: function(e, t) {
      e.exports = require('react');
    },
    1: function(e, t, n) {
      'use strict';
      Object.defineProperty(t, '__esModule', { value: !0 }), (t.getDataAttrs = void 0);
      var r = n(2);
      t.getDataAttrs = function(e) {
        return e
          ? (0, r.mapKeys)(e, function(e, t) {
              return 'data-' + (0, r.kebabCase)(t);
            })
          : {};
      };
    },
    2: function(e, t) {
      e.exports = require('lodash');
    },
    3: function(e, t) {
      e.exports = require('classnames');
    },
    5: function(e, t, n) {
      'use strict';
      function r(e) {
        return e && e.__esModule ? e : { default: e };
      }
      Object.defineProperty(t, '__esModule', { value: !0 });
      var o =
          Object.assign ||
          function(e) {
            for (var t = 1; t < arguments.length; t++) {
              var n = arguments[t];
              for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
            }
            return e;
          },
        a = n(0),
        c = r(a),
        i = n(1),
        l = n(7),
        u = r(l),
        s = function(e) {
          var t = e.icon,
            n = e.color,
            r = e.data;
          return c.default.createElement(
            'div',
            o(
              {
                className: u.default.wrapper,
                style: { fill: n },
                dangerouslySetInnerHTML: { __html: t },
              },
              (0, i.getDataAttrs)(r)
            )
          );
        };
      (s.defaultProps = { color: '#000', size: 16 }),
        (s.displayName = 'Plasma@Icon'),
        (t.default = s);
    },
    60: function(e, t, n) {
      'use strict';
      function r(e) {
        return e && e.__esModule ? e : { default: e };
      }
      function o(e, t, n) {
        return (
          t in e
            ? Object.defineProperty(e, t, {
                value: n,
                enumerable: !0,
                configurable: !0,
                writable: !0,
              })
            : (e[t] = n),
          e
        );
      }
      function a(e, t) {
        if (!(e instanceof t)) throw new TypeError('Cannot call a class as a function');
      }
      function c(e, t) {
        if (!e)
          throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return !t || ('object' != typeof t && 'function' != typeof t) ? e : t;
      }
      function i(e, t) {
        if ('function' != typeof t && null !== t)
          throw new TypeError(
            'Super expression must either be null or a function, not ' + typeof t
          );
        (e.prototype = Object.create(t && t.prototype, {
          constructor: { value: e, enumerable: !1, writable: !0, configurable: !0 },
        })),
          t && (Object.setPrototypeOf ? Object.setPrototypeOf(e, t) : (e.__proto__ = t));
      }
      Object.defineProperty(t, '__esModule', { value: !0 });
      var l =
          Object.assign ||
          function(e) {
            for (var t = 1; t < arguments.length; t++) {
              var n = arguments[t];
              for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
            }
            return e;
          },
        u = (function() {
          function e(e, t) {
            for (var n = 0; n < t.length; n++) {
              var r = t[n];
              (r.enumerable = r.enumerable || !1),
                (r.configurable = !0),
                'value' in r && (r.writable = !0),
                Object.defineProperty(e, r.key, r);
            }
          }
          return function(t, n, r) {
            return n && e(t.prototype, n), r && e(t, r), t;
          };
        })(),
        s = n(2),
        f = n(0),
        d = r(f),
        p = n(3),
        b = r(p),
        m = n(61),
        _ = r(m),
        h = n(5),
        x = r(h),
        y = n(1),
        v = (function(e) {
          function t() {
            return (
              a(this, t), c(this, (t.__proto__ || Object.getPrototypeOf(t)).apply(this, arguments))
            );
          }
          return (
            i(t, e),
            u(t, [
              {
                key: 'render',
                value: function() {
                  var e,
                    t,
                    n = this.props,
                    r = n.checked,
                    a = n.data,
                    c = n.description,
                    i = n.disabled,
                    u = n.fancy,
                    f = n.icon,
                    p = n.indeterminate,
                    m = n.isBold,
                    h = n.name,
                    v = n.onBlur,
                    k = n.onChange,
                    C = n.text,
                    w = n.value,
                    O = r || w,
                    g = (0, s.uniqueId)('id'),
                    j = (0, b.default)(o({}, _.default.inline, u)),
                    E = p ? _.default.indeterminate : _.default.original,
                    P = (0, b.default)(
                      _.default.wrapper,
                      ((e = {}),
                      o(e, _.default.wrapperDisabled, i),
                      o(e, _.default.fancy, u),
                      o(e, _.default.fancyChecked, u && O),
                      e)
                    ),
                    D = (0, b.default)(
                      _.default.text,
                      ((t = {}), o(t, _.default.textBold, m), o(t, _.default.textDisabled, i), t)
                    );
                  return d.default.createElement(
                    'div',
                    l({}, (0, y.getDataAttrs)(a), { className: j }),
                    d.default.createElement(
                      'div',
                      null,
                      d.default.createElement(
                        'label',
                        { htmlFor: g, className: P },
                        d.default.createElement('input', {
                          checked: O,
                          className: E,
                          disabled: i,
                          id: g,
                          name: h,
                          onChange: k,
                          onBlur: v,
                          type: 'checkbox',
                        }),
                        !u && d.default.createElement('div', { className: _.default.checkbox }),
                        f &&
                          d.default.createElement(
                            'div',
                            { className: u ? _.default.fancyIcon : _.default.icon },
                            d.default.createElement(x.default, { icon: f })
                          ),
                        d.default.createElement('div', { className: D }, C)
                      )
                    ),
                    c && d.default.createElement('span', { className: _.default.description }, c)
                  );
                },
              },
            ]),
            t
          );
        })(d.default.Component);
      (v.defaultProps = {
        checked: !1,
        indeterminate: !1,
        name: 'checkbox',
        text: 'Option',
        value: !1,
      }),
        (v.displayName = 'Plasma@Checkbox'),
        (t.default = v);
    },
    61: function(e, t) {
      e.exports = {
        text: 'src-components-Checkbox-text__17XBM',
        description: 'src-components-Checkbox-description__1-816',
        clearfix: 'src-components-Checkbox-clearfix__2irHj',
        inline: 'src-components-Checkbox-inline__1JEED',
        wrapper: 'src-components-Checkbox-wrapper__3JmR1',
        original: 'src-components-Checkbox-original__3CBDl',
        intermediate: 'src-components-Checkbox-intermediate__3bNEk',
        checkbox: 'src-components-Checkbox-checkbox__3nwSj',
        fancy: 'src-components-Checkbox-fancy__1M2lj',
        fancyChecked: 'src-components-Checkbox-fancyChecked__CF2m3',
        wrapperDisabled: 'src-components-Checkbox-wrapperDisabled__2Gx22',
        indeterminate: 'src-components-Checkbox-indeterminate__3MFpa',
        textBold: 'src-components-Checkbox-textBold__2H8NI',
        textDisabled: 'src-components-Checkbox-textDisabled__1tJX6',
        icon: 'src-components-Checkbox-icon__1C2sv',
        fancyIcon: 'src-components-Checkbox-fancyIcon__2hKYH',
      };
    },
    7: function(e, t) {
      e.exports = {
        clearfix: 'src-components-Icon-clearfix__3zhE-',
        wrapper: 'src-components-Icon-wrapper__1y5Fo',
      };
    },
  })
);
