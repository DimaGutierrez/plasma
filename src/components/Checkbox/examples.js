(function() {
  return {
    'Unchecked': <Checkbox text='I agree to all of it' />,
    'Checked': <Checkbox checked text='I agree to all of it' />,
    'Indeterminate': <Checkbox indeterminate text='Chicago' />,
  };
})
