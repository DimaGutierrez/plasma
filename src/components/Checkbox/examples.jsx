/* eslint-disable */
(function() {
  return {
    'Unchecked': <Checkbox text='I agree to all of it' />,
    'Checked': <Checkbox checked text='I agree to all of it' />,
    'Indeterminate unchecked': <Checkbox indeterminate text='Chicago' />,
    'Indeterminate checked': <Checkbox indeterminate checked text='Chicago' />,
  };
})
