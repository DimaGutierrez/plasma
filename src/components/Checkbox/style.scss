@import '../../styles/base';

.wrapper {
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
}

.original {
  position: absolute;
  opacity: 0;
  z-index: 10;
  cursor: pointer;
  &:checked ~ .faux {
    &::before {
      content: url('data:image/svg+xml;utf8,<svg width="12px" height="10px" viewBox="2 5 12 10" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs></defs><polygon id="Rectangle-34" stroke="none" fill="#282829" fill-rule="evenodd" points="6 15 2 10.8333333 3.6 9.16666667 6 11.6666667 12.4 5 14 6.66666667"></polygon></svg>');
    }
  }
}

.faux {
  width: 14px;
  height: 14px;
  background: none;
  border: 1px solid $cBorder;
  border-radius: 3px;
  background: $cWhite0;
  transition: border-color 0.1s ease-out;
  &::before {
    display: block;
    position: absolute;
    content: '';
    top: 1px;
    left: 2px;
    background: none;
  }
}

.text {
  margin-left: 10px;
  @extend %type--body;
  cursor: pointer;
}

