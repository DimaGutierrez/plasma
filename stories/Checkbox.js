import React from 'react';
import { storiesOf } from '@storybook/react';

import StoryState from '../StoryState';
import Checkbox from '../src/components/Checkbox/Checkbox';

storiesOf('Checkbox', module)
  .add('default', () => (
    <Checkbox />
  ))
  .add('with description', () => (
    <Checkbox description={'Hint: this is some text.'} />
  ))
  .add('checked', () => (
    <Checkbox checked />
  ))
  .add('unchecked', () => (
    <Checkbox checked={false} />
  ))
  .add('indeterminate unchecked', () => (
    <Checkbox indeterminate />
  ))
  .add('indeterminate checked', () => (
    <Checkbox indeterminate checked />
  ))
  .add('fancy', () => (
    <StoryState render={(state, setState) => (
      <Checkbox
        fancy
        name="optionOne"
        onChange={e => setState({selected: e.target.name})}
        checked={state.selected === 'optionOne'}
      />
    )}
    />
  ))
  .add('multiple fancy', () => (
    <div>
      <StoryState render={(state, setState) => (
        <Checkbox
          fancy
          name="dog"
          onChange={e => setState({selected: e.target.name})}
          text="Dog"
          checked={state.selected === 'dog'}
        />
      )}
      />
      <StoryState render={(state, setState) => (
        <Checkbox
          fancy
          name="cat"
          onChange={e => setState({selected: e.target.name})}
          text="Cat"
          checked={state.selected === 'cat'}
        />
      )}
      />
      <StoryState render={(state, setState) => (
        <Checkbox
          fancy
          name="snake"
          onChange={e => setState({selected: e.target.name})}
          text="Snake"
          checked={state.selected === 'snake'}
        />
      )}
      />
      <StoryState render={(state, setState) => (
        <Checkbox
          fancy
          name="bird"
          onChange={e => setState({selected: e.target.name})}
          text="Bird"
          checked={state.selected === 'bird'}
        />
      )}
      />
    </div>
  ));
;
