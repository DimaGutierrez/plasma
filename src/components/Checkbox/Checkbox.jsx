import { uniqueId } from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import style from './style.scss';
import { getDataAttrs, getDataProps } from '../../dataUtils';

class Checkbox extends React.Component {
  render() {
    const {
      checked,
      data,
      disabled,
      indeterminate,
      isBold,
      name,
      onBlur,
      onChange,
      text,
      value,
    } = this.props;
    const id = uniqueId('id');
    const inputClassName = indeterminate ? style.indeterminate : style.original;
    const wrapperClassName = cx(style.wrapper, {
      [style.wrapperDisabled]: disabled,
    });
    const textClassName = cx(style.text, {
      [style.textBold]: isBold,
      [style.textDisabled]: disabled,
    });

    return (
      <div {...getDataAttrs(data)}>
        <div>
          <label htmlFor={id} className={wrapperClassName}>
            <input
              checked={checked || value}
              className={inputClassName}
              disabled={disabled}
              id={id}
              name={name}
              onChange={onChange}
              onBlur={onBlur}
              type="checkbox"
            />
            <div className={style.checkbox} />
            <div className={textClassName}>{text}</div>
          </label>
        </div>
        {this.props.description && (
          <span className={style.description}>{this.props.description}</span>
        )}
      </div>
    );
  }
}

Checkbox.defaultProps = {
  checked: false,
  indeterminate: false,
  name: 'checkbox',
  text: 'Option',
  value: false,
};

Checkbox.propTypes = {
  checked: PropTypes.bool,
  description: PropTypes.string,
  disabled: PropTypes.bool,
  indeterminate: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.element]).isRequired,
  value: PropTypes.bool,
  ...getDataProps(),
};

Checkbox.displayName = 'Plasma@Checkbox';

export default Checkbox;
