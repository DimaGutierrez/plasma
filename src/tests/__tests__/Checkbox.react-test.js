/* eslint-disable */

import React from 'react';
import Checkbox from '../../components/Checkbox/Checkbox';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Checkbox Component', () => {
  test('render', () => {
    const component = renderer.create(
      <Checkbox />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.checked = true;
    expect(tree).toMatchSnapshot();
  });

  test('no fancy style', () => {
    const component = shallow(<Checkbox>Test Checkbox</Checkbox>);
    expect(component.is('.inline')).toBe(false);
  });

  test('fancy style', () => {
    const component = shallow(<Checkbox fancy>Test Fancy Checkbox</Checkbox>);
    expect(component.is('.inline')).toBe(true);
  });
});
