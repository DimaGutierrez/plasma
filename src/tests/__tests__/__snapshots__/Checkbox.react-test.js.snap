// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component render 1`] = `
<label
  className="wrapper"
  htmlFor="id1"
>
  <input
    checked={false}
    className="original"
    disabled={undefined}
    id="id1"
    name="checkbox"
    onChange={undefined}
    type="checkbox"
  />
  <div
    className="checkbox"
  />
  <div
    className="text"
  >
    Option
  </div>
</label>
`;

exports[`Checkbox Component render 2`] = `
<label
  className="wrapper"
  htmlFor="id1"
  isChecked={true}
>
  <input
    checked={false}
    className="original"
    disabled={undefined}
    id="id1"
    name="checkbox"
    onChange={undefined}
    type="checkbox"
  />
  <div
    className="checkbox"
  />
  <div
    className="text"
  >
    Option
  </div>
</label>
`;
